flink-connector/src/main/java/com/datastrato/gravitino/flink/connector/hive/GravitinoHiveCatalogFactory.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.flink.connector.hive;
+
+import com.datastrato.gravitino.flink.connector.utils.PropertyUtils;
+import java.util.Set;
+import org.apache.flink.configuration.ConfigOption;
+import org.apache.flink.table.catalog.Catalog;
+import org.apache.flink.table.catalog.hive.HiveCatalog;
+import org.apache.flink.table.catalog.hive.factories.HiveCatalogFactory;
+import org.apache.flink.table.catalog.hive.factories.HiveCatalogFactoryOptions;
+import org.apache.flink.table.factories.CatalogFactory;
+import org.apache.flink.table.factories.FactoryUtil;
+import org.apache.hadoop.hive.conf.HiveConf;
+
+/**
+ * Factory for creating instances of {@link GravitinoHiveCatalog}. It will be created by SPI
+ * discovery in Flink.
+ */
+public class GravitinoHiveCatalogFactory implements CatalogFactory {
+  /** Identifier for the {@link GravitinoHiveCatalog}. */
+  public static final String IDENTIFIER = "gravitino-hive";
+
+  private HiveCatalogFactory hiveCatalogFactory;
+
+  @Override
+  public Catalog createCatalog(Context context) {
+    this.hiveCatalogFactory = new HiveCatalogFactory();
+    final FactoryUtil.CatalogFactoryHelper helper =
+        FactoryUtil.createCatalogFactoryHelper(this, context);
+    helper.validateExcept(
```

Comment:
This will skip the validation with the specified prefix.

Reply:
What if user add a properties like `flink.bypass.aa` by mistake, this will cause all Flink jobs fail?   I prefer to ignore it.

Reply:
@FANNG1 It's good for users. How about just removing the validation? Otherwise, I have to remove these useless key before the validation.

Reply:
@FANNG1 I thought about it again. The validation is needed, it can be used to validate the required options such as `metastore.uris`. Let me check how to finish it.

Reply:
please add comment before `helper.validateExcept` to descript how it works